### ISSUE TITLE
feat: route middleware `after`

### ISSUE THREAD

**Issue by dargmuesli** [nuxt/nuxt#23982]
### Describe the feature

The Nuxt docs [say](https://nuxt.com/docs/api/composables/use-router#navigation-guards):

> useRouter composable provides afterEach, beforeEach and beforeResolve helper methods that acts as navigation guards.
> 
> However, Nuxt has a concept of route middleware that simplifies the implementation of navigation guards and provides a better developer experience.

I wonder how `afterEach` could be implemented using navigation guards. As far as I know they only run *before* navigation.

If that is not possible right now, would this make up for a neat feature request?

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by posva** [comment]
I don't think we need to introduce a new abstraction for this as it can already be implemented with a plugin

```ts
  // plugins/router-after-each.ts
  export default defineNuxtPlugin(() => {
    const router = useRouter()
    router.afterEach((to, from) => {
      // your "after" logic here
    })
  })
```

Note: a regular plugin registers its `afterEach` *after* the initial navigation has already resolved, so the hook will only fire on subsequent navigations. If you need to react to the initial route too, handle it explicitly at plugin setup (e.g. read `router.currentRoute.value`).
